src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -0,0 +1,560 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ControlInteractivity.h"
+#include <argb.h>
+#include <DefaultSettings.h>
+#include <unicode.hpp>
+#include <Utf16Parser.hpp>
+#include <Utils.h>
+#include <WinUser.h>
+#include <LibraryResources.h>
+#include "../../types/inc/GlyphWidth.hpp"
+#include "../../types/inc/Utils.hpp"
+#include "../../buffer/out/search.h"
+
+#include "ControlInteractivity.g.cpp"
+
+using namespace ::Microsoft::Console::Types;
+using namespace ::Microsoft::Console::VirtualTerminal;
+using namespace ::Microsoft::Terminal::Core;
+using namespace winrt::Windows::Graphics::Display;
+using namespace winrt::Windows::System;
+using namespace winrt::Windows::ApplicationModel::DataTransfer;
+
+namespace winrt::Microsoft::Terminal::Control::implementation
+{
+    ControlInteractivity::ControlInteractivity(IControlSettings settings,
+                                               TerminalConnection::ITerminalConnection connection) :
+        _touchAnchor{ std::nullopt },
+        _lastMouseClickTimestamp{},
+        _lastMouseClickPos{},
+        _selectionNeedsToBeCopied{ false }
+    {
+        _core = winrt::make_self<ControlCore>(settings, connection);
+    }
+
+    void ControlInteractivity::UpdateSettings()
+    {
+        _updateSystemParameterSettings();
+    }
+
+    void ControlInteractivity::Initialize()
+    {
+        _updateSystemParameterSettings();
+
+        // import value from WinUser (convert from milli-seconds to micro-seconds)
+        _multiClickTimer = GetDoubleClickTime() * 1000;
+    }
+
+    winrt::com_ptr<ControlCore> ControlInteractivity::GetCore()
+    {
+        return _core;
+    }
+
+    // Method Description:
+    // - Returns the number of clicks that occurred (double and triple click support).
+    // Every call to this function registers a click.
+    // Arguments:
+    // - clickPos: the (x,y) position of a given cursor (i.e.: mouse cursor).
+    //    NOTE: origin (0,0) is top-left.
+    // - clickTime: the timestamp that the click occurred
+    // Return Value:
+    // - if the click is in the same position as the last click and within the timeout, the number of clicks within that time window
+    // - otherwise, 1
+    unsigned int ControlInteractivity::_numberOfClicks(winrt::Windows::Foundation::Point clickPos,

```

Comment:
This is totally internal, right? I thought I saw one in TermControl but my memory is hazy

Reply:
yea, this is gone from `TermControl`